// @mantine/tiptap@7.2.2/cjs/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var RichTextEditor = require('./RichTextEditor.js');
var RichTextEditor_context = require('./RichTextEditor.context.js');
var labels = require('./labels.js');
var RichTextEditorControlsGroup = require('./RichTextEditorControlsGroup/RichTextEditorControlsGroup.js');
var RichTextEditorControl = require('./RichTextEditorControl/RichTextEditorControl.js');
var RichTextEditorContent = require('./RichTextEditorContent/RichTextEditorContent.js');
var Link = require('./extensions/Link.js');
var RichTextEditorLinkControl = require('./RichTextEditorControl/RichTextEditorLinkControl.js');
var RichTextEditorColorPickerControl = require('./RichTextEditorControl/RichTextEditorColorPickerControl.js');
var RichTextEditorColorControl = require('./RichTextEditorControl/RichTextEditorColorControl.js');
var controls = require('./RichTextEditorControl/controls.js');



exports.RichTextEditor = RichTextEditor.RichTextEditor;
exports.useRichTextEditorContext = RichTextEditor_context.useRichTextEditorContext;
exports.DEFAULT_LABELS = labels.DEFAULT_LABELS;
exports.RichTextEditorControlsGroup = RichTextEditorControlsGroup.RichTextEditorControlsGroup;
exports.RichTextEditorControl = RichTextEditorControl.RichTextEditorControl;
exports.RichTextEditorContent = RichTextEditorContent.RichTextEditorContent;
exports.Link = Link.Link;
exports.RichTextEditorLinkControl = RichTextEditorLinkControl.RichTextEditorLinkControl;
exports.RichTextEditorColorPickerControl = RichTextEditorColorPickerControl.RichTextEditorColorPickerControl;
exports.RichTextEditorColorControl = RichTextEditorColorControl.RichTextEditorColorControl;
exports.AlignCenterControl = controls.AlignCenterControl;
exports.AlignJustifyControl = controls.AlignJustifyControl;
exports.AlignLeftControl = controls.AlignLeftControl;
exports.AlignRightControl = controls.AlignRightControl;
exports.BlockquoteControl = controls.BlockquoteControl;
exports.BoldControl = controls.BoldControl;
exports.BulletListControl = controls.BulletListControl;
exports.ClearFormattingControl = controls.ClearFormattingControl;
exports.CodeBlockControl = controls.CodeBlockControl;
exports.CodeControl = controls.CodeControl;
exports.H1Control = controls.H1Control;
exports.H2Control = controls.H2Control;
exports.H3Control = controls.H3Control;
exports.H4Control = controls.H4Control;
exports.H5Control = controls.H5Control;
exports.H6Control = controls.H6Control;
exports.HighlightControl = controls.HighlightControl;
exports.HrControl = controls.HrControl;
exports.ItalicControl = controls.ItalicControl;
exports.OrderedListControl = controls.OrderedListControl;
exports.StrikeThroughControl = controls.StrikeThroughControl;
exports.SubscriptControl = controls.SubscriptControl;
exports.SuperscriptControl = controls.SuperscriptControl;
exports.UnderlineControl = controls.UnderlineControl;
exports.UnlinkControl = controls.UnlinkControl;
exports.UnsetColorControl = controls.UnsetColorControl;
